### PR TITLE
Port TLDDoc Source Code 

### LINKED ISSUE #195: Port TLDDoc Code to JSTL-API Project
While we can't move to [VLDDoc](https://github.com/eclipse-ee4j/jstl-api/issues/169), we can import the TLDDoc project code into jstl-api project after approval: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23881 

This would allow up to make the necessary updates to the TLDDoc to avoid various issues (javax references & TLD EE10 namespaces)..

### PR DESCRIPTION
**Pull request by volosied**
fixes #195

Approval was granted via QC 23881, so I'm checking the code in. 

- Added tlddoc module to the main pom (project is build via `mvn clean install` when run at root) 
- _Impl_ will use the locally build tlddoc jar (2.0-SNAPSHOT)
- Updates source/target to 1.8 (min needs to be 1.6, however)

I used the default pom.xml, with a few major modifications: 
- Updated groupId, ArtifactId, and version.
- Removed description, organization, and inception year 

Source: https://repo1.maven.org/maven2/taglibrarydoc/tlddoc/1.3

### CODE REVIEW

**Comment by volosied** [comment]
@pnicolucci  @arjantijms  I skipped the deployment step so the TLDDoc jar shouldn't be published (but will be used locally when building jstl-api) 

Let me know if you have any concerns / questions here.

File: tlddoc/pom.xml [hunk 1]
```diff
@@ -38,6 +37,14 @@ Tag Library Documentation Generator is a utility for automatically generating ja
                     <skip>true</skip>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <release>11</release>
```
Comment (pzygielo) [inline tlddoc/pom.xml:44]: With this set here, perhaps `maven.compiler.source/target` properties could be dropped?

Comment (volosied) [reply tlddoc/pom.xml:44]: Ah, you're right. Thank you, let me update

File: tlddoc/pom.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <name>taglibrarydoc</name>
+    <groupId>taglibrarydoc</groupId>
+    <version>2.0-SNAPSHOT</version>
+    <url>http://taglibrarydoc.dev.java.net</url>
+    <licenses>
+        <license>
+            <name>BSD License</name>
+            <url>
+http://taglibrarydoc.dev.java.net/source/browse/*checkout*/taglibrarydoc/taglibrarydoc/license.txt
+            </url>
```
Comment (pnicolucci) [inline tlddoc/pom.xml:14]: Does this link actually work for you?

File: tlddoc/pom.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                <version>3.8.1</version>
+                <configuration>
+                    <release>11</release>
+                    <compilerArgument>-Xlint:unchecked</compilerArgument>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
Comment (pnicolucci) [inline tlddoc/pom.xml:47]: New line please

File: tlddoc/pom.xml (hunk 2 above)
Comment (volosied) [reply tlddoc/pom.xml:14]: Didn't think too much about the URL.... Looks like a bad link.  Should I remove the URL part entirely? Or keep it as it is?

Closest item I can find so far is : http://web.archive.org/web/20070808225932/http://www.opensource.org/osi3.0/licenses/bsd-license.php  
- Linked from http://web.archive.org/web/20070624180825/https://taglibrarydoc.dev.java.net/

Comment (pnicolucci) [reply tlddoc/pom.xml:14]: http://web.archive.org/web/20070624180825/https://taglibrarydoc.dev.java.net/ points to : http://web.archive.org/web/20070808225932/http://www.opensource.org/osi3.0/licenses/bsd-license.php

So I think we're safe to point to: http://opensource.org/licenses/BSD-3-Clause

Comment (volosied) [reply tlddoc/pom.xml]: I agree.  Both list 3 clauses, we should be good.  

I've updated the pom.

**Review by pnicolucci — APPROVED** [review]

**Pull request merged**